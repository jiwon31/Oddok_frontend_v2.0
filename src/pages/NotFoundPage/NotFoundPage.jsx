import React from "react";
import { Link } from "react-router-dom";
import styles from "./NotFoundPage.module.css";

function NotFoundPage() {
  return (
    <div className={styles.container}>
      <div className={styles.title}>요청하신 페이지를 찾을 수 없습니다.</div>
      <Link to="/">메인으로 돌아가기🏠</Link>
    </div>
  );
}

export default NotFoundPage;
