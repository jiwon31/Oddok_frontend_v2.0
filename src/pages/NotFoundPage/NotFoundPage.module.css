.container {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: white;
}

.button {
  padding: 0.8rem 1rem;
  border-radius: 4px;
  font-size: 1rem;
  font-weight: 600;
  color: var(--gray950);
  background-color: var(--primary);
}

.button:hover {
  filter: brightness(1.1);
}
