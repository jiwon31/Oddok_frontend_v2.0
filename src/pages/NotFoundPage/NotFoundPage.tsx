import { useNavigate } from "react-router-dom";
import styles from "./NotFoundPage.module.css";

export default function NotFoundPage() {
  const navigate = useNavigate();

  return (
    <div className={styles.container}>
      <h2>요청하신 페이지를 찾을 수 없습니다.</h2>
      <button className={styles.button} type="button" onClick={() => navigate(-1)}>
        이전 화면으로 돌아가기
      </button>
    </div>
  );
}
