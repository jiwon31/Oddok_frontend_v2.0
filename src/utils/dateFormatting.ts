// Date 객체를 yyyy-mm-dd 형식의 문자열로 반환
export default function dateFormatting(date: Date): string {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  return `${year}-${month < 10 ? `0${month}` : month}-${day < 10 ? `0${day}` : day}`;
}
