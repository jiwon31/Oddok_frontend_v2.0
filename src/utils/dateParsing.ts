// yyyy-mm-dd 형태의 문자열을 Date 객체로 변환
export default function dateParsing(value: string): Date {
  const date = new Date(value); // 로컬 타임존이 반영된 시간
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(date.getTime() + offset);
}
