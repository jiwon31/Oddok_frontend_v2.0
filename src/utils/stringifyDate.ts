export default function stringifyDate(date: Date): string {
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(+date - offset).toISOString();
}
