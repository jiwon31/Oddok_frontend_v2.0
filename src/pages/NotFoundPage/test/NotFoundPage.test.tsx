import { Link, Route } from "react-router-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { withRouter } from "tests/utils";
import NotFoundPage from "../NotFoundPage";

describe("NotFound Page", () => {
  it("renders", () => {
    const component = renderer.create(withRouter(<Route path="/" element={<NotFoundPage />} />));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("버튼을 누르면 이전 페이지로 돌아간다.", async () => {
    function PreviousPage() {
      return <Link to="*">Go to NotFound</Link>;
    }

    render(
      withRouter(
        <>
          <Route path="/" element={<PreviousPage />} />
          <Route path="*" element={<NotFoundPage />} />
        </>,
      ),
    );

    userEvent.click(screen.getByRole("link"));
    expect(await screen.findByText(/이전 화면으로 돌아가기/i)).toBeInTheDocument();
    userEvent.click(screen.getByRole("button"));
    expect(await screen.findByText(/Go to NotFound/i)).toBeInTheDocument();
  });
});
