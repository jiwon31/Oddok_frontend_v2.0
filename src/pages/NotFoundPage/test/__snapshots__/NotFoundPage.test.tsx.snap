// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page renders 1`] = `
<div
  className="container"
>
  <h2>
    요청하신 페이지를 찾을 수 없습니다.
  </h2>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    이전 화면으로 돌아가기
  </button>
</div>
`;
